jfr-microbench/src/main/java/io/netty/handler/jfr/microbench/NoopHandler.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.jfr.microbench;
+
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+
+import java.net.SocketAddress;
+
+public class NoopHandler extends ChannelDuplexHandler {
+
```

Comment:
Add a private constructor and expose the instance as static one to reduce object allocations. This also means you will need to override `isSharable()` and return `true`.

Reply:
This goes for JfrEventHandler as well I guess?

Reply:
yes as long as has not state it can be sharable

Reply:
But we do want it do have state I guess, reusing the events per channel

Reply:
Yep if we do per Channel we should create a new handler per channel